calico/felix/frules.py

```diff
@@ -198,61 +182,52 @@ def install_global_rules(config, v4_filter_updater, v6_filter_updater,
 
     # Ensure that Calico-controlled IPv6 hosts cannot spoof their IP addresses.
     # (For IPv4, this is controlled by a per-interface sysctl.)
+    v6_raw_prerouting_chain, v6_raw_prerouting_deps = \
+        config.plugins["iptables_generator"].raw_rpfilter_failed_chain(
+            ip_version=6)
+
+    v6_raw_updater.rewrite_chains({CHAIN_PREROUTING: v6_raw_prerouting_chain},
+                                  {CHAIN_PREROUTING: v6_raw_prerouting_deps},
+                                  async=False)
+
     v6_raw_updater.ensure_rule_inserted(
-        "PREROUTING --in-interface %s --match rpfilter --invert -j DROP" %
-        iface_match,
-        async=False,
-    )
+        "PREROUTING --in-interface %s --match rpfilter --invert "
+        "--jump %s" %
+        (iface_match, CHAIN_PREROUTING),
+        async=False)
 
     # The IPV4 nat table first. This must have a felix-PREROUTING chain.
-    nat_pr = []
-    if config.METADATA_IP is not None:
-        # Need to expose the metadata server on a link-local.
-        #  DNAT tcp -- any any anywhere 169.254.169.254
-        #              tcp dpt:http to:127.0.0.1:9697
-        nat_pr.append("--append " + CHAIN_PREROUTING + " "
-                      "--protocol tcp "
-                      "--dport 80 "
-                      "--destination 169.254.169.254/32 "
-                      "--jump DNAT --to-destination %s:%s" %
-                      (config.METADATA_IP, config.METADATA_PORT))
-    v4_nat_updater.rewrite_chains({CHAIN_PREROUTING: nat_pr}, {}, async=False)
+    # Write the chain first and then udpate the v4 NAT kernel chain to
+    # reference it.
+    prerouting_chain, prerouting_deps = \
+        config.plugins["iptables_generator"].nat_prerouting_chain(ip_version=4)
```

Comment:
I'd introduce a local for `config.plugins["iptables_generator"]` jsut to keep the code DRY (and shorten a lot of these lines).  PyCharm has a refactoring command for that if that's what you're using